types/facepaint/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for facepaint 1.2
+// Project: https://github.com/emotion-js/facepaint
+// Definitions by: Ciarán Curley <https://github.com/DogPawHat>
+//                 Anton Samper Rivaya <https://github.com/antonsamper>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+export type BaseArg = object | object[];
```

Comment:
The shipped code of `facepaint` exports a single value, a function, and has no `default` property [1], which is required by the module spec. So you need to match that with an `export =` instead of `export default`. It also means that the exported types will need to be put into a namespace:

```ts
declare namespace facepaint {
    export type BaseArg ...
    // others ...
}
declare function facepaint(breakpoints: facepaint.Selector[], options?: facepaint.Options): facepaint.DynamicStyleFunction;
export = facepaint;
```

And will need to be imported like so:

```ts
import facepaint = require('facepaint');
```

[1]

I tried this on npm.runkit.com:

```js
var f = require('facepaint')
console.log(f.default)
// prints undefined
```

Reply:
The package can be used by `import default` actually:
https://github.com/emotion-js/facepaint/blob/master/src/index.js#L2
https://github.com/emotion-js/facepaint/blob/master/package.json#L6

Reply:
Only the "module" property points to a version with a default. `"main": "dist/index.cjs.js"` does not, which means that the following code would be broken without --esModuleInterop:

```ts
// @Filename: myfile.ts
import x from 'facepaint';
console.log(x)
```

becomes

```js
// @Filename: myfile.js
var x = require('facepaint').default
console.log(x)
```

And

```sh
$ node myfile.js
undefined
```